NumberBaseball/NumberBaseball.c

```diff
@@ -1,20 +1,141 @@
 #include <stdio.h>
+#include <stdlib.h>
 
-void Init()
+#define LINESIZE (1024)
+
+void Init(int *number1, int *number2, int *number3)
 {
+	int test1;
+	int test2;
+	int test3;
+	while (1)
+	{
+		test1 = rand() % 10;
+		test2 = rand() % 10;
+		test3 = rand() % 10;
+
+		if (test1 == test2 || test1 == test3 || test2 == test3)
+		{
+			continue;
+		}
+
+		*number1 = test1;
+		*number2 = test2;
+		*number3 = test3;
+		break;
+	}
+	printf("Welcome to Number Baseball!\n");
+}
+
+void Update(int *number1, int *number2, int *number3) {
+	int guess1;
+	int guess2;
+	int guess3;
+	while (1) {
+		char line[LINESIZE];
+		int numbersEntered = 0;
+		guess1 = -1;
+		guess2 = -1;
+		guess3 = -1;
+		printf("Choose 3 numbers:\n");
+		if (!fgets(line, LINESIZE, stdin)) {
+			clearerr(stdin);
+			printf("\n");
+			continue;
+		}
+		if ((numbersEntered = sscanf(line, "%d %d %d",
+			&guess1, &guess2, &guess3)) == 3)
+		{
+			int correctCount = 0;
+
+			if (guess1 == *number1)
+			{
+				printf("Strike! ");
+				correctCount++;
+			}
+			else if (guess1 == *number2 || guess1 == *number3)
+			{
+				printf("Out! ");
+			}
+			else
+			{
+				printf("Ball! ");
+			}
+
+			if (guess2 == *number2)
+			{
+				printf("Strike! ");
+				correctCount++;
+			}
+			else if (guess2 == *number1 || guess2 == *number3)
+			{
+				printf("Out! ");
+			}
+			else
+			{
+				printf("Ball! ");
+			}
+
+			if (guess3 == *number3)
+			{
+				printf("Strike!\n");
+				correctCount++;
+			}
+			else if (guess3 == *number1 || guess3 == *number2)
+			{
+				printf("Out!\n");
+			}
+			else
+			{
+				printf("Ball!\n");
+			}
+
+			if (correctCount == 3) {
+				printf("You win!\n");
+				exit(1);
```

Comment:
Use return

Reply:
But if i use return that just brings me out to the while loop that starts updating again. Should i move the contents of Init() into Update() to keep playing the game by picking new numbers for the computer?

Reply:
1. Init() just only initializes data for the game.
2.If it keeps updating, why dont u try modifying while loop condition in main?

Reply:
so in this case should i use malloc (etc) or is this too simple for now

Reply:
> so in this case should i use malloc (etc) or is this project too simple to merit that

You don't even need to use malloc for the project